qa/statsd/src/cases/06.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env pmpython
+# -*- coding: utf-8 -*-
+
+# Exercises gauge metric type
+
+import sys
+import socket
+import glob
+import os
+
+utils_path = os.path.abspath(os.path.join("utils"))
+sys.path.append(utils_path)
+
+import pmdastatsd_test_utils as utils
+
+utils.print_test_file_separator()
+print(os.path.basename(__file__))
+
+ip = "0.0.0.0"
+port = 8125
+sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+
+payloads = [
+    # thrown away
+    "-1wqeqe",
+    "-20weqe0",
+    "-wqewqe20",
+    # ok
+    "0",
+    "1",
+    "10", 
+    "100",
+    "1000",
+    "10000",
+    "+1",
+    "+10",
+    "+100",
+    "+1000",
+    "+10000",
+    "-0.1",
+    "-0.01",
+    "-0.001",
+    "-0.0001",
+    "-0.00001"
+]
+
+basis_parser_config = os.path.join("configs", "single", "parser_type", "0", "pmdastatsd.ini")
+ragel_parser_config = os.path.join("configs", "single", "parser_type", "1", "pmdastatsd.ini")
+
+duration_aggregation_basic_config = os.path.join("configs", "single", "duration_aggregation_type", "0", "pmdastatsd.ini")
+duration_aggregation_hdr_histogram_config = os.path.join("configs", "single", "duration_aggregation_type", "1", "pmdastatsd.ini")
+
+testconfigs = [basis_parser_config, ragel_parser_config, duration_aggregation_basic_config, duration_aggregation_hdr_histogram_config]
+
+def run_test():
+    for testconfig in testconfigs:
+        utils.print_test_section_separator()
+        utils.pmdastatsd_install(basis_parser_config)
+        for payload in payloads:
+            sock.sendto("test_gauge:{}|g".format(payload).encode("utf-8"), (ip, port))
+        utils.print_metric("statsd.pmda.dropped")
+        utils.print_metric('statsd.test_gauge')
+        overflow_payload = sys.float_info.max
+        sock.sendto("test_gauge2:+{}|g".format(overflow_payload).encode("utf-8"), (ip, port))
+        utils.print_metric("statsd.pmda.dropped")
+        utils.print_metric("statsd.test_gauge2")
+        sock.sendto("test_gauge2:+{}|g".format(overflow_payload).encode("utf-8"), (ip, port))
+        utils.print_metric("statsd.pmda.dropped")
+        utils.print_metric("statsd.test_gauge2")
+        # TODO: check if this is truly the desired behavior
```

Comment:
Now would be a good time to check, and figure out an answer (before this is merged) - otherwise, we're all likely to forget over time.

Reply:
I actually already did the other day (we had small convo in Slack about it), emailed it trough with Lukas if it is OK and we resolved it. I will remove the comment in code.